lib/active_model_serializers/adapter/json_api/resource_identifier.rb

```diff
@@ -20,11 +20,13 @@ def as_json
 
         def type_for(serializer)
           return serializer._type if serializer._type
-          if ActiveModelSerializers.config.jsonapi_resource_type == :singular
-            serializer.object.class.model_name.singular
-          else
-            serializer.object.class.model_name.plural
+          type = serializer.object.class.to_s.split('::')
+          type.map! { |t| t.underscore.downcase }
```

Comment:
I just checked KeyTransform and I don't see how it could help here. Correct me if I'm wrong, but it transforms keys in supplied hash, and we're only working with string here. Any suggestions how to apply it here?